src/libs/collect.c

```diff
@@ -1304,21 +1304,22 @@ static void list_view(dt_lib_collect_rule_t *dr)
         g_free(makermodel_query);
         break;
 
-      case DT_COLLECTION_PROP_HISTORY: // History, 2 hardcoded alternatives
+      case DT_COLLECTION_PROP_HISTORY: // History
+        // images without history are counted as if they were basic
         g_snprintf(query, sizeof(query),
                    "SELECT CASE"
-                   "       WHEN initial_hash == current_hash THEN '%s'"
+                   "       WHEN basic_hash == current_hash THEN '%s'"
                    "       WHEN auto_hash == current_hash THEN '%s'"
                    "       WHEN current_hash IS NOT NULL THEN '%s'"
                    "       ELSE '%s'"
                    "     END as altered, 1, COUNT(*) AS count"
                    " FROM main.images AS mi"
-                   " LEFT JOIN (SELECT DISTINCT imgid, initial_hash, auto_hash, current_hash"
+                   " LEFT JOIN (SELECT DISTINCT imgid, basic_hash, auto_hash, current_hash"
                    "            FROM main.history_hash) ON id = imgid"
                    " WHERE %s"
                    " GROUP BY altered"
                    " ORDER BY altered ASC",
-                    ("initial"), _("auto applied"), _("altered"), _("none"), where_ext);
+                   _("basic"), _("auto applied"), _("altered"), _("basic"), where_ext);
```

Comment:
2 times basic here ? Is that expected?

Reply:
yes, here we want to retrieve both the images without history + images which have current = basic hash into the same basic group.
Now, there is another way: to have a group "never opened" (= no history) separated from basic (current = basic hash). "Never opened" group should match "order=none". This is more factual but I'm not sure that will help the end user.
Let me know if you thinks this last way is better.

Reply:
I'm for "never opened" being present.

Reply:
"never opened" stops being factual if you have disabled embedded thumnails

Reply:
> "never opened" stops being factual if you have disabled embedded thumbnails

This setting is not what one changes everyday. If not set, "never opened" tells the truth (and suggests the user to open the images) otherwise it doesn't appear.
Would be "not developed"  better (more accurate but less suggestive) ?

Reply:
"not developed" seems like better alternative, because as you've said it's more accurate.

Reply:
Isn't "basic" (and actually "auto applied") also "not developed"? What use-case do you have in mind? And why would you want to add a feature that only works when embedded thumbnails are enabled?

Reply:
Not sure to understand your concern.

> Isn't "basic" (and actually "auto applied") also "not developed"? What use-case do you have in mind? 

As is we have:
- basic = `no history` (= no order = no hash = not developed) + `current_hash = basic_hash` (= only mandatory modules (then developed but not altered))
- auto applied = `current_hash = auto_hash ` (then develop but not altered)
- altered = current_hash != basic_hash and current_hash != auto_hash 

The discussion here is just to decide if it is worth to split the first case. We would have then:
- not developed (appears only if enabled embedded thumbnails) potentially basic or auto-applied
- basic (mandatory module only)
- auto applied (mandatory module + auto applied ones)
- altered

> And why would you want to add a feature that only works when embedded thumbnails are enabled?

The fact is the current master makes confusion about no history and not altered in particular when disabled embedded thumbnails. The goal of this PR is to solve the inconsistencies listed in #4313 , in particular those introduced by the import optimization which doesn't create the corresponding history, either masking (basic stays basic and doesn't move to altered if no change) that for the user (current state of the PR) or giving him the necessary information (never opened or not developed) to understand what does happen. 

I agree that neither "never opened" and "not developed" are perfect and can lead to inadequate interpretation. What would your proposal ?
For the time being I'd prefer slightly the first solution because the user may not be interested by the import history optimization. If it was the case he could use the "module order" collection, where he can see the "none" case (=no history...).

Reply:
My question is how each of the mentioned cases are useful when using darktable. Otherwise it's just exposing internal states and complicating the UI.

Reply:
There is still another way: create the history at import time ! In that case the former discussion has no object any more.

Reply:
The usage might stem from module order collection being a bit wonky due to inconsistencies around history and then history being wonky due to fact that some developments were considered developed and some not.

I have couple of usecases for history (and module order) being:
1. i want list of images i haven't developed at all (just imported)
2. i want list of images that have auto-apply presets applied since I'm most sure I messd something with them.
3. i want list of developed (or altered) images (even if that's just white balance, which previously didn't affect history/altered icon), because those are most likely candidates for export
4. i want list of images with custom module order, because those are most likelly caused by my (wonky) mouse dragging stuff and in 99% of cases those might be messed edits and i need to fix those
5. i want list of images with older moduler order, so i could probably upgrade those to new order.

all those use cases and more are covered by changes by @phweyland :)

Reply:
So we have:

1. just imported
2. with just auto-applied presets
3. altered

and

a. old module order
b. new default module order
c. modified module order

I don't see here "none"/"no history"/"not developed" or whatever.

Reply:
>     1. just imported

This is not correct.

Reply:
What is not correct? From user point of view there are 3 actions one can do:
1. import
2. open in darkroom
3. edit (change settings)

What else is there?

Reply:
@TurboGit 
I don't see any @parafin points which would be misaligned with the current state of the PR.

I feel he is concerned with the split of basic group into "never opened" or "not developed" and "basic" one, but this is unclear for me.

Anyway, both ways are fine for me. Please let me know how to follow up on this.

Reply:
I'm just advocating for having some high-level design notes (e.g. usage scenarios we want to cover) and working at implementing them. The fact that "embedded thumbnails" preference affects history creation isn't something we want to expose to users. Simple way of thinking about this is to imagine how you would describe a feature in user documentation. How do you explain the difference between "never opened" and "basic" without it becoming overly complicated?

Reply:
@parafin: You missed one action from user's point of view, the fact that some auto-applied presets have been triggered or not. This is really a user's oriented action as auto-presets are added by users. So to me I still see 3 possibles states from user's point of view, but those:

1. imported (with possibly some required iop set like temperature for RAW) 
2. with auto-applied presets
3. manually edited

As discussed before (IIRC) the fact that we go into darkrroom is not an action and should not be counted if there is no modification done. Now the part that may be confusing is that we are also trying to mix information about iop-order (legacy or v3.0) and this order can be set by points 2 and point 3.

I think imported state above is what you said to be "basic", right?

Reply:
I'm OK with this classification, but it's still only 3 states, not 4;)

Reply:
@parafin ; If you're referring to "never opened" and "never developed" I agree with you. This won't make sense to end user's I suppose.

@phweyland : Most of the usages I see (and let's concentrate only on this for this PR, we'll see later if we want to split status) is to select quickly the non-edited images, the ones that need some work to be done. And of course the status altered from lighttable which is the "manually edited" (in darkroom or using copy/paste or style) state described above.

I think we are at this stage currently, right?

Reply:
> I think we are at this stage currently, right?

Yes, that's the current stage (3 states).

Reply:
> I'm just advocating for having some high-level design notes (e.g. usage scenarios we want to cover) and working at implementing them. The fact that "embedded thumbnails" preference affects history creation isn't something we want to expose to users. Simple way of thinking about this is to imagine how you would describe a feature in user documentation. How do you explain the difference between "never opened" and "basic" without it becoming overly complicated?

I agree with that. Saying that at the beginning would simplify the conversation, wouldn't it ?

Reply:
So let me test this again as it is now and merge if all goes well.